src/functions/module_filter.ts

```diff
@@ -1,9 +1,95 @@
+import { cellsSortingCriteria } from "../helpers/sort";
 import { _t } from "../translation";
-import { AddFunctionDescription, Arg, isMatrix, Matrix, Maybe, ValueAndFormat } from "../types";
+import {
+  AddFunctionDescription,
+  Arg,
+  CellValue,
+  CellValueType,
+  isMatrix,
+  Locale,
+  Matrix,
+  Maybe,
+  ValueAndFormat,
+} from "../types";
 import { NotAvailableError } from "../types/errors";
 import { arg } from "./arguments";
 import { assertSameDimensions, assertSingleColOrRow } from "./helper_assert";
-import { assert, matrixMap, toBoolean, toMatrix, transposeMatrix } from "./helpers";
+import { toScalar } from "./helper_matrices";
+import {
+  assert,
+  matrixMap,
+  toBoolean,
+  toMatrix,
+  toNumber,
+  toString,
+  transposeMatrix,
+} from "./helpers";
+
+function sortMatrix(
+  _range: Matrix<ValueAndFormat>,
+  locale: Locale,
+  ...values: (ValueAndFormat | Matrix<ValueAndFormat>)[]
+): Matrix<ValueAndFormat> {
+  const criteriaIndex: (0 | 1)[] = [];
```

Comment:
the purpose of this list of ones and zeros is not straight forward to understand.
Why not booleans ? or even full strings for a better understanding what it means.
A ternary expression can be used instead of array indexing `sortingCriteria` (by 0 or 1)